mypy/checkexpr.py

```diff
@@ -106,6 +110,7 @@ def __init__(self,
         """Construct an expression type checker."""
         self.chk = chk
         self.msg = msg
+        self.type_context = [UninhabitedType()]
```

Comment:
It's `Optional[Type]` anyway since `accept` may add `None` values to the type context (but we don't check mypy using `--strict-optional` so it's unchecked). `UninhabitedType` is probably okay, but we use `None` to mark an empty type context elsewhere, so that is a bit inconsistent. There would also be a small risk of breaking something. Changing the type annotations of `type_context` to `List[Optional[Type]]` would be reasonable, though.